src/tools/kata-ctl/src/ops/volume_ops.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022 Boston University
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::args::{DirectVolSubcommand, DirectVolumeCommand};
+
+use anyhow::{anyhow, Ok, Result};
+use futures::executor;
+use hyper;
+use kata_types::mount::DirectVolumeMountInfo;
+use std::{fs, path::Path, path::PathBuf, time::Duration};
+use url;
+
+use runtimes;
+
+const TIMEOUT: Duration = Duration::new(5, 0);
+const CONTENT_TYPE_JSON: &str = "application/json";
+const MOUNT_INFO_FILE_NAME: &str = "mountInfo.json";
+const KATA_DIRECT_VOLUME_ROOT_PATH: &str = "/run/kata-containers/shared/direct-volumes";
+
+pub fn handle_direct_volume(vol_cmd: DirectVolumeCommand) -> Result<()> {
+    let command = vol_cmd.directvol_cmd;
+    match command {
+        DirectVolSubcommand::Add(args) => {
+            add(&args.volume_path, &args.mount_info)?;
+        }
+        DirectVolSubcommand::Remove(args) => {
+            remove(&args.volume_path)?;
+        }
+        DirectVolSubcommand::Stats(args) => {
+            let result = executor::block_on(stats(&args.volume_path))?;
+            println!("{}", &result);
+        }
+        DirectVolSubcommand::Resize(args) => {
+            executor::block_on(resize(&args.volume_path, args.resize_size))?;
+        }
+    }
+    Ok(())
+}
+
+async fn resize(volume_path: &str, size: u64) -> Result<()> {
```

Comment:
All paths (in fact all parameters) should be treated as suspicious by default - see comments below.

Reply:
I think for `stats` and `resize` commands which send HTTP requests to runtime-rs, it would be better if the input validation happend on the `runtime-rs` side, since a malicious user would still be able to forge a malicious request on the client side.